specification.md

```diff
@@ -286,12 +286,12 @@ Our expression function definitions can now be referenced by workflow states whe
      "dataConditions": [
         {
           "name": "Applicant is adult",
-          "condition": "{{ fn(isAdult) }}",
+          "condition": "${ fn:isAdult }",
```

Comment:
yes, we need a language specific mapping to tell parsers/validators that its a call to an expression functions so they substitute its "operation" value that includes the actual expression that needs to be evaluated. "fn" is just a namespace in this case.

the data the expression is evaluated against depends on if its used inside a certain filter, in a state property or in a top-level workflow property. 
I will look at possibility to pass in the input specifically, for example "fn:isAdult(.customer)". Will test it out and do that in a separate pr if possible. Right now its assumed its always "fn:expressionFunctionName(.)" but users dont have to do the "(.)" part - is assumed

Reply:
I think what bothers me about this is we need to inject special behavior into the middle of the jq expressions in order to call these functions, it's no longer JQ at this point but some strange hybrid language.  This puts a burden on users of the workflow language because they have to learn this new expression language special to calling workflow functions, and I worry that it actually makes the runtimes more difficult to implement.

It seems like if we have the expectation that all of the workflows expression 'functions' are automatically exposed as jq functions by the runtime, then we shouldn't need a special syntax to tell the runtime to 'substitute' the call.  The caveat would be that, if some of the expression functions aren't jq then they would not be accessible from jq expressions (which imho, is a reasonable approach).

Given the example above, it seems like the runtime under-the-covers could define the expression functions as jq functions before evaluating a condition.

Given this alternative way to define this line:
``` "condition": "${isAdult(.)}"```

Then runtime could do something functionally equivalent to the follow to evaluate the expression:

```
def isAdult: .applicant | .age >= 18;
def isMinor: in.applicant | .age < 18;
isAdult(.)
```

Reply:
ugh sorry i went ahead and merged before realizing to see this comment
i will open an issue on this now and work on it for next week